Source/Scene/Model.js

```diff
@@ -250,6 +253,43 @@ define([
         return animationIds;
     }
 
+    function getPerInstanceUniforms(cachedGltf) {
+        // Find all the uniforms that are updated per instance. Usually these are uniforms that include a model
+        // matrix. They will be changed to take an instanced vertex attribute instead.
+        var perInstanceUniforms = {};
+        if (defined(cachedGltf) && defined(cachedGltf.gltf)) {
+            var techniques = cachedGltf.gltf.techniques;
+            for (var techniqueName in techniques) {
+                if (techniques.hasOwnProperty(techniqueName)) {
+                    var technique = techniques[techniqueName];
+                    var parameters = technique.parameters;
+                    var pass = technique.passes[technique.pass];
+                    var instanceProgram = pass.instanceProgram;
+                    var programName = instanceProgram.program;
+                    // Different techniques may share the same program, skip if already processed.
+                    // This assumes techniques that share a program do not declare different semantics for the same uniforms.
+                    if (!defined(perInstanceUniforms[programName])) {
+                        var uniformMap = {};
+                        perInstanceUniforms[programName] = uniformMap;
+                        var uniforms = instanceProgram.uniforms;
+                        for (var uniformName in uniforms) {
+                            if (uniforms.hasOwnProperty(uniformName)) {
+                                var parameterName = uniforms[uniformName];
+                                var parameter = parameters[parameterName];
+                                var semantic = parameter.semantic;
+                                if (defined(semantic) && (semantic.indexOf('MODEL') > -1)) {
```

Comment:
I don't think this is the most forward-compatibility thing to do.  Instead, let's explicitly look for the known `MODEL` semantics.  Here is the list: https://github.com/KhronosGroup/glTF/issues/83#issuecomment-140797614